plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -71,14 +71,20 @@ def run(self, params={}, state={}, custom_config: dict = {}):  # pylint: disable
                 f"A PluginException has occurred. Status code {status_code} returned. Error: {error}. "
                 f"Existing state: {existing_state}"
             )
-            return [], existing_state, False, status_code, error
+            return [], existing_state, False, status_code, error.data
```

Comment:
What's the context behind this change?

Is error.data a `PluginException`? I don't think we should be returning whatever is in error.data here

Reply:
When we return `error.data`, this is what we get
```
{
    "body": {
        "has_more_pages": false,
        "log": "Connect: Connecting...\nrapid7/Palo Alto Cortex XDR:4.0.0. Step name: monitor_alerts\nNo previous alert time within state.\nApplying the following start time='2024-09-24T09:59:19.000000Z'.\nStarting to download alerts...\nQuery start time: 2024-09-24T09:59:19.000000Z, Query end time: 2024-09-25T09:59:19.000000Z\nA PluginException has occurred. Status code 401 returned. Error: An error occurred during plugin execution!\n\nAuthentication failed: invalid credentials. Please verify the credentials for your account and try again.. Existing state: {}\n",
        "meta": {},
        "output": [],
        "state": {},
        "status": "ok",
        "status_code": 401
    },
    "type": "task_event",
    "version": "v1"
}
```

Reply:
It may work in this specific scenario, but we shouldn't be returning response data where it should be an Exception

Reply:
Okay good to know, I can add that in now